testing/e2e/skr-tester/pkg/command/update.go

```diff
@@ -0,0 +1,194 @@
+		RunE:    func(_ *cobra.Command, _ []string) error { return cmd.Run() },
+	}
+	cmd.cobraCmd = cobraCmd
+
+	cobraCmd.Flags().StringVarP(&cmd.instanceID, "instanceID", "i", "", "InstanceID of the specific instance.")
+	cobraCmd.Flags().StringVarP(&cmd.planID, "planID", "p", "", "PlanID of the specific instance.")
+	cobraCmd.Flags().BoolVarP(&cmd.updateMachineType, "updateMachineType", "m", false, "Should update machineType.")
+
+	return cobraCmd
+}
+
+func (cmd *UpdateCommand) Run() error {
+	cmd.log = logger.New()
+	brokerClient := broker.NewBrokerClient(broker.NewBrokerConfig())
+	catalog, err := brokerClient.GetCatalog()
+	if err != nil {
+		return fmt.Errorf("failed to get catalog: %v", err)
+	}
+	services, ok := catalog["services"].([]interface{})
+	if !ok {
+		return errors.New("services field not found or invalid in catalog")
+	}
+	for _, service := range services {
+		serviceMap, ok := service.(map[string]interface{})
+		if !ok {
+			return errors.New("service is not a map[string]interface{}")
+		}
+		if cmd.updateMachineType {
+			currentMachineType, err := getCurrentMachineType(cmd.instanceID)
+			if err != nil {
+				return fmt.Errorf("failed to get current machine type: %v", err)
+			}
+			fmt.Printf("Current machine type: %s\n", *currentMachineType)
+			if serviceMap["id"] != "47c9dcbf-ff30-448e-ab36-d3bad66ba281" {
+				continue
+			}
+			plans, ok := serviceMap["plans"].([]interface{})
+			if !ok {
+				return errors.New("plans field not found or invalid in serviceMap")
+			}
+			for _, p := range plans {
+				planMap, ok := p.(map[string]interface{})
+				if !ok || planMap["id"] != cmd.planID {
+					continue
+				}
+				updateParams, err := extractUpdateParams(planMap)
+				if err != nil {
+					return fmt.Errorf("failed to extract update parameters: %v", err)
+				}
+				if len(updateParams) < 2 {
+					continue
+				}
+				for i, m := range updateParams {
+					if m == *currentMachineType {
+						newMachineType := updateParams[(i+1)%len(updateParams)].(string)
+						fmt.Printf("Determined machine type to update: %s\n", newMachineType)
+						resp, err := brokerClient.UpdateInstance(cmd.instanceID, map[string]interface{}{"machineType": newMachineType})
+						if err != nil {
+							return fmt.Errorf("error updating instance: %v", err)
+						}
+						fmt.Printf("Update operationID: %s\n", resp["operation"].(string))
+						break
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (cmd *UpdateCommand) Validate() error {
+	if cmd.instanceID != "" && cmd.planID != "" {
+		return nil
+	} else {
+		return errors.New("you must specify the planID and instanceID")
+	}
+}
+
+func getCurrentMachineType(instanceID string) (*string, error) {
+	_, err := exec.Command("kcp", "login").Output()
+	if err != nil {
+		return nil, fmt.Errorf("failed to run kcp login: %v", err)
+	}
+	output, err := exec.Command("kcp", "rt", "-i", instanceID, "-ojson", "--runtime-config").Output()
```

Comment:
Try the following command: `kcp rt -i <INSTANCE_ID> --runtime-config -o custom=":{.runtimeConfig.spec.shoot.provider.workers[0].machine.type}"`
You should get a response with a empty line (empty header) and a machine type.

Reply:
Tried it but the `workers[0]` is not working

Reply:
Changed.

Reply:
Nice 👍